### ISSUE TITLE
Namespace parameter in read_ncbi_gene2go is poorly implemented

### ISSUE THREAD

**Issue by Melnorme1984** [tanghaibao/goatools#134]
The `read_ncbi_gene2go` function in `associations.py` only supports using one namespace at a time in normal usage. That seems silly - why not make that parameter a list instead of a single value?

Meanwhile if you supply more than one taxonomy id or use the `taxid2asscs` keyword, the namespace parameter seems to be ignored entirely. Not good!

**Comment by Melnorme1984** [comment]
Ah, I see that you've recently made changes related to this: https://github.com/tanghaibao/goatools/issues/131

The current implementation still seems not ideal, though.

**Comment by dvklopfenstein** [comment]
Hello, 

Thanks for your thoughts. `read_ncbi_gene2go` was an older function that existed while all of our annotation format readers were managed independently.

Each of our four annotation readers was added by request, with the requests spaced years apart, resulting in each annotation reader being completely independent from the others.

This no longer made sense when asked by users to add control for including and excluding annotations using user-specified evidence code and evidence code class. So I added an annotation reader base class and four derived classes for each of the four formats that we support. Upon doing this, I replaced the old code with our new implementation. And that is why it `read_ncbi_gene2go` clunky.

So it is time to deprecate the old function. We will keep it deprecated for a long time before removing it so people have plenty of time to switch or ask questions, if needed. 

Thank you for taking the time to write and share your thoughts.
